cores/cv32e40p/user_manual/source/debug.rst

```diff
@@ -70,6 +80,49 @@ Several trigger registers are required to adhere to specification. The following
 The TDATA1.DMODE is hardwired to a value of 1. In non Debug Mode,
 writes to Trigger registers are ignored and reads reflect CSR values.
 
+Debug state
+-----------
+
+As specified in `RISC-V Debug Specification <https://riscv.org/specifications/debug-specification/>`_ every hart that can be selected by
+the Debug Module is in exactly one of four states: ``nonexistent``, ``unavailable``, ``running`` or ``halted``. Harts are ``nonexistent`` if
+they will never be part of the system, no matter how long a user waits. Harts are ``unavailable`` if they might become available at a later
+time, or if there are other harts with higher indexes than this one. Harts may be ``unavailable`` for a variety of reasons including being
+reset or temporarily being powered down.  Harts are ``running`` when they are executing normally, as if no debugger was attached. This
+includes being in a low power mode or waiting for an interrupt, as long as a halt request will result in the hart being ``halted``.
+Harts are ``halted`` when they are in Debug Mode, only performing tasks on behalf of the debugger.
```

Comment:
I would keep the first sentence but remove the rest of the paragraph as this is explained in the debug spec. I'm OK leaving it here as well as this is a clear definition for immediate reference.

Reply:
Done